WordPress/Classes/Services/SharingService.swift

```diff
@@ -451,25 +455,14 @@ public class SharingService : LocalCoreDataService
     // MARK : Private Instance Methods
 
 
-    /// Returns the API to use with the service
+    /// Returns the API to use with the service. 
     ///
-    private func apiForRequest() -> WordPressComApi {
-        var api: WordPressComApi? = nil
-
-        if let restApi = AccountService(managedObjectContext: managedObjectContext).defaultWordPressComAccount()?.restApi {
-            api = restApi.hasCredentials() ? restApi : nil
-        }
-
-        if api == nil {
-            // In practice this should never happen, but if it does let's try to detect it. 
-            // Write to the error log if the api was nil, and trigger an assert to 
-            // catch this in development/QA.
-            api = WordPressComApi.anonymousApi()
-            let error = "SharingService is not using a real WordPress.com account."
-            DDLogSwift.logError(error)
-            assert(false, error)
-        }
-
+    /// - Parameters:
+    ///     - blog: The blog to use for the rest api.
+    ///
+    private func apiForBlog(blog: Blog) -> WordPressComApi {
+        let api: WordPressComApi? =  blog.restApi()
+        assert(api != nil)
         return api!
```

Comment:
> we add nullability annotations to Blog [...] we'll be able to avoid the cast to optional.

I'll happily remove the optional cast when that happens!  For now I think the extra clarity the deliberate unwrapping provides is a good thing.  Thanks again :)